packages/js/data/src/products/types.ts

```diff
@@ -45,79 +45,80 @@ export type Product< Status = ProductStatus, Type = ProductType > = Omit<
 	Schema.Post,
 	'status' | 'categories'
 > & {
-	id: number;
-	name: string;
-	slug: string;
-	permalink: string;
+	attributes: ProductAttribute[];
+	average_rating: string;
+	backordered: boolean;
+	backorders: 'no' | 'notify' | 'yes';
+	backorders_allowed: boolean;
+	button_text: string;
+	categories: Pick< ProductCategory, 'id' | 'name' | 'slug' >[];
 	date_created: string;
 	date_created_gmt: string;
 	date_modified: string;
 	date_modified_gmt: string;
-	type: Type;
-	status: Status;
-	featured: boolean;
-	description: string;
-	short_description: string;
-	sku: string;
 	date_on_sale_from_gmt: string | null;
 	date_on_sale_to_gmt: string | null;
-	virtual: boolean;
+	description: string;
+	dimensions: ProductDimensions;
+	download_expiry: number;
+	download_limit: number;
 	downloadable: boolean;
 	downloads: ProductDownload[];
-	download_limit: number;
-	download_expiry: number;
 	external_url: string;
-	button_text: string;
-	tax_status: 'taxable' | 'shipping' | 'none';
-	tax_class: 'standard' | 'reduced-rate' | 'zero-rate' | undefined;
-	manage_stock: boolean;
-	stock_quantity: number;
+	featured: boolean;
+	id: number;
 	low_stock_amount: number;
-	stock_status: 'instock' | 'outofstock' | 'onbackorder';
-	backorders: 'no' | 'notify' | 'yes';
+	manage_stock: boolean;
+	name: string;
+	on_sale: boolean;
+	permalink: string;
 	price: string;
 	price_html: string;
+	purchasable: boolean;
 	regular_price: string;
+	rating_count: number;
+	related_ids: number[];
+	reviews_allowed: boolean;
```

Comment:
Sorry for the larger diff here.  This (`reviews_allowed`) was the only added item.

The remaining were alphabetized to make it a bit faster to find items.